script/write_config.py

```diff
@@ -76,6 +77,21 @@ def escape(s: str) -> str:
             s[i] = "\\s"
     return "".join(s)
 
+# Some values need to be "transformed", such as being converted from hex to decimal
+def transformValues(d_orig):
+    d = d_orig.copy()
+
+    c_path = os.path.expanduser("~/.config/kcminputrc")
+    if c_path in d:
+        for item_str in list(d[c_path]):
+            if item_str.startswith("Libinput/"):
+                item = d[c_path].pop(item_str)
+                item_str_list = item_str.split("/")
+                item_str_list[1] = str(int(item_str_list[1],16))
+                item_str_list[2] = str(int(item_str_list[2],16))
+                item_str = "/".join(item_str_list)
+                d[c_path][item_str] = item
+    return d;
```

Comment:
I don't really like this because it changes the interface of the `configFile` option in an unexpected manner. It's fine if we want to have the options of the input module expect the ids in hex, but then the conversion should also be done in the module (possibly with something like [this](https://github.com/Misterio77/nix-colors/blob/b92df8f5eb1fa20d8e09810c03c9dc0d94ef2820/lib/core/conversions.nix#L72-L94)).

This change broke my config, which included `programs.plasma.configFile.kcminputrc."Libinput/1390/269/ELECOM TrackBall Mouse HUGE TrackBall".ScrollMethod.value =  4;`. Now the ids are silently mangled.

Reply:
I agree there. It should be fairly straight forward to do so as well. I'll try to do the hex-conversion in nix instead and open a PR about it.

Reply:
>  but then the conversion should also be done in the module

This would be indeed better.

Reply:
Should be fixed now after #295.